### DISCUSSION TITLE
[Section 4.18 - Data Sources] - Error launching source instance

### DISCUSSION THREAD

**Discussion by danielcgil83** [chgasparoto/curso-aws-com-terraform#43] [Q&A]
Olá Cleber, primeiramente quero parabenizar, o curso está excelente, muito claro e objetivo!

Me deparei com um erro ao tentar subir o serviço de EC2 nesta aula:

╷
│ Error: Error launching source instance: InvalidParameterValue: The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
│       status code: 400, request id: 1afdc8e7-ce23-437c-a861-d0e3243ab74d
│ 
│   with aws_instance.web,
│   on ec2.tf line 2, in resource "aws_instance" "web":
│    2: resource "aws_instance" "web" {
│ 
╵

Consegui fazer a implementação fazendo uma alteração no arquivo "data.tf":

```hcl
data "aws_ami" "ubuntu" {
  owners      = ["amazon"]
  most_recent = true
  name_regex  = "ubuntu"

  filter {
    name   = "architecture"
    values = ["x86_64"]
  }
}
```

Mesmo tendo conseguido implementar com esta alteração, fiquei com uma dúvida, esta foi a solução correta ou eu deveria ter feito de outra forma?

**Comment by chgasparoto** [comment]
Obrigado @danielcgil83 

Teve um outro comentário que outro aluno teve o mesmo problema. O provider da AWS ou a AWS mesmo atualizou o nome da AMI para "Ubuntu" com U maiúsculo. https://github.com/chgasparoto/curso-aws-com-terraform/discussions/26#discussioncomment-783776

Está correto a sua solução, eu fiz a mesma coisa neste arquivo. https://github.com/chgasparoto/curso-aws-com-terraform/blob/master/05-autoscaling-app/ec2.tf#L6

Abraço.
